### DISCUSSION TITLE
Try to eject installation disk automatically

### DISCUSSION THREAD

**Discussion by mikhailnov** [rhinstaller/anaconda#4193] [Ideas]
Hi

We in ROSA made such patch for Anaconda of the rhel-9 branch: https://abf.io/import/anaconda/blob/573b765478/0032-Try-to-eject-installation-disk-automatically.patch

I wanted to pull request it, but I see that post-scripts have been depreceated.

So I suggest to have a look at that code and implement the same if needed.

Bellow is the script that is executed after installing the system:

```bash
# Schedule eject of CD/DVD ROM if we have installed the system successfully.
# Some hyvervisors, e.g. VirtualBox, treat eject as removing the virtual drive.
# This will prevent further boot of the installed system from the installation media.
# Systemd will run tne created script after unmounting all other filesystems.
set -e
set -u
systemd_dir=""
if mount | grep -qE '^/dev/(sr|cd|dvd).*/run/initramfs/live' ; then
	if [ -d /usr/lib/systemd/system-shutdown ]; then
		systemd_dir=/usr/lib/systemd/system-shutdown
	elif [ -d /lib/systemd/system-shutdown ]; then
		systemd_dir=/lib/systemd/system-shutdown
	fi
	if [ -n "$systemd_dir" ] && [ -x /usr/bin/eject ]; then
		echo '#!/bin/sh' > "$systemd_dir"/eject-on-shutdown
		echo '/usr/bin/eject -m' >> "$systemd_dir"/eject-on-shutdown
		chmod +x "$systemd_dir"/eject-on-shutdown
	fi
fi
```

**Comment by VladimirSlavik** [comment]
Thanks! I wonder if you actually need that? The media should be ejected automatically in most cases. In short:
- If there is no kickstart at all involved, eject.
- If using `interactive-defaults.ks` eject.
- If using `OEMDRV` and `ks.cfg`, do not eject.
- If using external kickstart, do not eject.
- If set in kickstart with `reboot --eject`, override the negative cases above and eject.

See more details in [find_kickstart](https://github.com/rhinstaller/anaconda/blob/master/pyanaconda/startup_utils.py#L352) (sets `flags.eject`), [exitHandler](https://github.com/rhinstaller/anaconda/blob/master/anaconda.py#L88) (uses also kickstart data) and the eject itself in [dracut_eject](https://github.com/rhinstaller/anaconda/blob/master/pyanaconda/core/util.py#L515).

Based on that, you should be able to eject the media using just the kickstart, I think?
